rfcs/0188-batched-label-to-task-status.md

```diff
@@ -0,0 +1,83 @@
+# RFC 188 - Adding a batched "label to task status" endpoint
+* Comments: [#188](https://github.com/taskcluster/taskcluster-rfcs/pull/188)
+* Proposed by: @Alphare
+
+# Summary
+
+Add an endpoint to query the latest status for a given task label, for each
+label in a provided list.
+
+## Motivation
+
+When looking at profiles with the following command in the `mozilla-unified`
+repo:
+
+`py-spy record -F --idle --format speedscope -o output.json -- ./mach taskgraph morphed -p taskcluster/test/params/mc-onpush.yml`
+
+I saw that the majority (near 70%) of the time was spent on the optimization
+step, as the screenshot below shows:
+
+![20231211_10h54m47s_grim](https://github.com/taskcluster/taskcluster/assets/9445758/62c400cc-a125-4f08-b7dd-c8bc9a9e9a6d)
+
+Further investigating, it turns out that basically all of this time is spent
+waiting on the network doing `read()` calls. Basically, for every task, there
+is one call to `find_task_id`, then one call to `status_task`, the goal
+of which is to get the status of the latest task that corresponds to a label.
+In our case, this corresponds to 900+ API calls, incurring a massive performance
+cost to the client, and I'm guessing it's not negligible from the server's
+standpoint. (Why some of those calls take multiple seconds I have no idea,
+it might be worth investigating separately.)
+
+## Proof of concept
+
+To see if my intuition was correct, I've recorded the mapping of
+`label -> status` and replayed the command with the mapping acquired from a
+JSON file to simulate a (very) fast single API call that would give the
+right structure. The optimization step goes down to 2 seconds from 2+ minutes,
+as shown in the screenshot below:
+
+![20231211_10h55m01s_grim](https://github.com/taskcluster/taskcluster/assets/9445758/1849c8a1-fcc0-403b-acaf-ea997c875505)
+
+# Details
+
+Implementation details:
+
+- New API endpoint `/status`
+- HTTP GET:
+    - Request body consisting of a JSON object:
+        ```
+        {
+            "labels": <list of labels, not sorted, non-unique>
+        }
+        ```
+    - Response body:
+        ```
+        {
+            "mapping": {
+                <label>: <status of latest task for label>
+            },
+            "continuationToken": <continuation token>
```

Comment:
jcristau pointed out that if implementing a continuationToken is too hard, we could just set an arbitrary limit (e.g 1000) and leave it to consumers to chunk their requests.

As a consumer, I'd prefer a continuation token though :)